File: python/Figuras/UnidadMedida.py
"""
Enumeración para unidades de medida
"""
from enum import Enum
from typing import Dict


class UnidadMedida(Enum):
    """Enumeración de unidades de medida disponibles"""
    
    METROS = "m"
    CENTIMETROS = "cm"
    MILIMETROS = "mm"
    PULGADAS = "in"
    PIES = "ft"
    
    def get_factor_conversion(self) -> float:
        """
        Obtiene el factor de conversión a metros
        
        Returns:
            float: Factor de conversión a metros
        """
        factores = {
            UnidadMedida.METROS: 1.0,
            UnidadMedida.CENTIMETROS: 0.01,
            UnidadMedida.MILIMETROS: 0.001,
            UnidadMedida.PULGADAS: 0.0254,
            UnidadMedida.PIES: 0.3048
        }
        return factores[self]
    
    def get_simbolo(self) -> str:
        """
        Obtiene el símbolo de la unidad
        
        Returns:
            str: Símbolo de la unidad
        """
        return self.value
    
    def get_nombres_completos(self) -> Dict[str, str]:
        """
        Obtiene un diccionario con los nombres completos de las unidades
        
        Returns:
            Dict[str, str]: Diccionario con nombres completos
        """
        return {
            UnidadMedida.METROS.value: "metros",
            UnidadMedida.CENTIMETROS.value: "centímetros",
            UnidadMedida.MILIMETROS.value: "milímetros",
            UnidadMedida.PULGADAS.value: "pulgadas",
            UnidadMedida.PIES.value: "pies"
        }

File: java/Figuras/src/main/java/org/example/UnidadMedida.java
package org.example;

import java.util.HashMap;
import java.util.Map;

/**
 * Enumeración para unidades de medida
 */
public enum UnidadMedida {
    METROS("m"),
    CENTIMETROS("cm"),
    MILIMETROS("mm"),
    PULGADAS("in"),
    PIES("ft");
    
    private final String simbolo;
    
    /**
     * Constructor privado del enum
     * @param simbolo Símbolo de la unidad
     */
    UnidadMedida(String simbolo) {
        this.simbolo = simbolo;
    }
    
    /**
     * Obtiene el factor de conversión a metros
     * @return Factor de conversión a metros
     */
    public double getFactorConversion() {
        switch (this) {
            case METROS:
                return 1.0;
            case CENTIMETROS:
                return 0.01;
            case MILIMETROS:
                return 0.001;
            case PULGADAS:
                return 0.0254;
            case PIES:
                return 0.3048;
            default:
                return 1.0;
        }
    }
    
    /**
     * Obtiene el símbolo de la unidad
     * @return Símbolo de la unidad
     */
    public String getSimbolo() {
        return this.simbolo;
    }
    
    /**
     * Obtiene un mapa con los nombres completos de las unidades
     * @return Mapa con nombres completos
     */
    public static Map<String, String> getNombresCompletos() {
        Map<String, String> nombres = new HashMap<>();
        nombres.put(METROS.getSimbolo(), "metros");
        nombres.put(CENTIMETROS.getSimbolo(), "centímetros");
        nombres.put(MILIMETROS.getSimbolo(), "milímetros");
        nombres.put(PULGADAS.getSimbolo(), "pulgadas");
        nombres.put(PIES.getSimbolo(), "pies");
        return nombres;
    }
}